### PR TITLE
Notifications system

### PR DESCRIPTION
**Pull request by hamadodene**

### CODE REVIEW

File: backend/src/controllers/notificationController.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+const Notification = require('../models/notificationModel')
+
+//Handle get all notification
+exports.getNotifications = function(req, res) {
+    Notification.find({ user: req.params.user }, function(err, notifications) {
+        if (err) {
+            res.status(500).send(err);
```
Comment (fedpet) [inline backend/src/controllers/notificationController.js:7]: No json?

File: backend/src/controllers/notificationController.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            data: notifications
+        });
+    });
+}
+
+//Handle view get notification by id
+exports.getNotificationById = function(req, res) {
+    Notification.findById(req.params.notification_id, function(err, notification) {
+        if (err) {
+            res.status(500).send(err);
```
Comment (fedpet) [inline backend/src/controllers/notificationController.js:19]: json?

File: backend/src/controllers/notificationController.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        }
+        res.status(200).json({
+            data: notification
+        });
+    });
+}
+
+//Handle create new notification
+exports.createNotification = function(req, res) {
+    var nt = new notification();
```
Comment (fedpet) [inline backend/src/controllers/notificationController.js:29]: wait, how does notifications with lower case n work? Isn't javascript case sensitive?

File: backend/src/controllers/notificationController.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    nt.type = req.body.type;
+    nt.quantity = req.body.type;
+    nt.user = req.body.user;
+
+    nt.save(function(err) {
+        if (err) {
+            res.status(500).send(error);
```
Comment (fedpet) [inline backend/src/controllers/notificationController.js:36]: json

File: backend/src/routes.js [hunk 5]
```diff
@@ -3,15 +3,28 @@ const router = express.Router()
...
-router.use(auth.unless({path: ['/login']}))
+router.use(auth.unless({ path: ['/login'] }))
 
 router.post('/login', account.login)
 router.get('/account', account.account)
 
+//Notifications system
+router.route('/user/:user/notifications')
+    .get(notification.getNotifications)
+    .post(notification.createNotification);
```
Comment (fedpet) [inline backend/src/routes.js:16]: I don't think we should let the client create notifications with an endpoint. I think we should just create them internally when someone delivers waste

Comment (hamadodene) [reply backend/src/routes.js:16]: Make sense

File: backend/src/controllers/notificationController.js (hunk 3 above)
Comment (hamadodene) [reply backend/src/controllers/notificationController.js:29]: Sorry, I wanted to write **Notification**

File: backend/src/controllers/notificationController.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    });
+}
+
+//Handle mark notification as read
+exports.markAsRead = function(req, res) {
+    Notification.findById(req.params.notification_id, function(err, notification) {
+        if (err) {
+            res.status(500).json(err);
+        }
+        notification.status = true;
```
Comment (fedpet) [inline backend/src/controllers/notificationController.js:33]: what's `notification.status` ?

File: backend/src/models/notificationModel.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    },
+    type: {
+        type: String,
+        required: true
+    },
+    quantity: {
+        type: Number,
+        required: true
+    },
+    user: String
```
Comment (fedpet) [inline backend/src/models/notificationModel.js:21]: if this is meant to be a pointer to an user, maybe it should be `user: Schema.Types.ObjectId`

**Review by fedpet — CHANGES_REQUESTED** [review]

**Pull request closed without merging**